### PR TITLE
[gguf number of params] proposal of a cleaner implem

### PR DESCRIPTION
**Pull request by julien-c**
i propose the following tweaks to #637

Let me know what you think @mishig25

### CODE REVIEW

**Comment by julien-c** [comment]
ping @coyotte508 @mishig25 for a quick review:)

File: packages/gguf/src/types.ts [hunk 1]
```diff
@@ -138,5 +138,5 @@ export interface GGUFTensorInfo {
 export interface GGUFParseOutput {
 	metadata: GGUFMetadata;
 	tensorInfos: GGUFTensorInfo[];
-	paramCount?: number;
+	parameterCount?: number;
```
Comment (mishig25) [inline packages/gguf/src/types.ts:141]: this comment applies https://github.com/huggingface/huggingface.js/pull/637#discussion_r1581189423

File: packages/gguf/src/gguf.ts [hunk 2]
```diff
@@ -177,14 +178,33 @@ function readMetadataValue(
...
+export async function gguf(
+	url: string,
+	params: {
+		/**
+		 * Custom fetch function to use instead of the default one, for example to use a proxy or edit headers.
+		 */
+		fetch?: typeof fetch;
+		computeParametersCount: true;
+	}
+): Promise<SetRequired<GGUFParseOutput, "parameterCount">>;
```
Comment (mishig25) [inline packages/gguf/src/gguf.ts:190]: nice usage of type-fest!

**Review by mishig25 — APPROVED** [review]
except https://github.com/huggingface/huggingface.js/pull/639#discussion_r1582805197, lgtm !

**Review by coyotte508 — APPROVED** [review]

File: packages/gguf/src/types.ts (hunk 1 above)
Comment (mishig25) [reply packages/gguf/src/types.ts:141]: yes

Comment (mishig25) [reply packages/gguf/src/types.ts:141]: `GGUFParseOutput ` stays as it is.

The return of the function should look like: `GGUFParseOutput  & { parameterCount?: number }`

Comment (julien-c) [reply packages/gguf/src/types.ts:141]: ok i'll tweak this 🙏

Comment (julien-c) [reply packages/gguf/src/types.ts]: 322af6bab0de2099df3f622688e8e22df07176a5

**Pull request merged**